cloudnet/src/main/java/de/dytanic/cloudnet/command/commands/CommandModules.java

```diff
@@ -1,138 +1,253 @@
+                                        .map(IModuleWrapper::getModuleConfiguration)
+                                        .map(ModuleConfiguration::getName)
+                                        .distinct()
+                                        .collect(Collectors.toList())
+                        )
+                )
+
+                .getSubCommands(), "modules", "module"
+        );
+
+        super.prefix = "cloudnet";
+        super.permission = "cloudnet.command." + names[0];
+        super.description = LanguageManager.getMessage("command-description-" + names[0]);
     }
 
-    /**
-     * An unsafe implementation for managing the module system
-     * Use the "reload" for a save module reloading
-     *
-     * @param sender
-     * @param command
-     * @param args
-     * @param commandLine
-     * @param properties
-     */
-    @Deprecated
-    @Override
-    public void execute(ICommandSender sender, String command, String[] args, String commandLine, Properties properties) {
-        IModuleProvider moduleProvider = CloudNetDriver.getInstance().getModuleProvider();
-        Collection<IModuleWrapper> moduleWrappers = moduleProvider.getModules();
-
-        if (args.length == 0) {
-            sender.sendMessage(
-                    "Modules(" + moduleWrappers.size() + "): " + Arrays.toString(moduleWrappers.stream()
-                            .map(moduleWrapper -> moduleWrapper.getModule().getName()).toArray(String[]::new)),
-                    " ",
-                    "modules list | group=<name> name=<name> version=<version>"
-            );
-            return;
+    private static boolean matchesProperties(ModuleId moduleId, Properties properties) {
+        if (properties.containsKey("group") && !moduleId.getGroup().contains(properties.get("group"))) {
+            return false;
         }
+        if (properties.containsKey("name") && !moduleId.getName().contains(properties.get("name"))) {
+            return false;
+        }
+        return !properties.containsKey("version") || moduleId.getVersion().contains(properties.get("version"));
+    }
 
-        if (args[0].equalsIgnoreCase("list")) {
-            for (IModuleWrapper wrapper : moduleWrappers) {
-                if (properties.containsKey("group") && !wrapper.getModuleConfiguration().getGroup().contains(properties.get("group"))) {
-                    continue;
-                }
-                if (properties.containsKey("name") && !wrapper.getModuleConfiguration().getName().contains(properties.get("name"))) {
-                    continue;
-                }
-                if (properties.containsKey("version") && !wrapper.getModuleConfiguration().getVersion().contains(properties.get("version"))) {
-                    continue;
-                }
+    private static void displayRemoteModuleInfo(ICommandSender sender, RepositoryModuleInfo moduleInfo) {
```

Comment:
Codacy found an issue: [The method 'displayRemoteModuleInfo(ICommandSender, RepositoryModuleInfo)' has an NPath complexity of 729, current threshold is 200](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5772469)